secure_message_v2/views/messages.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+from flask import Blueprint, Response, jsonify, make_response, request
+from sqlalchemy.exc import IntegrityError, StatementError
+from structlog import wrap_logger
+from werkzeug.exceptions import BadRequest, NotFound
+
+from secure_message_v2.controllers.messages import create_message
+from secure_message_v2.controllers.validate import Exists, Validator
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+messages_bp = Blueprint("messages_bp", __name__)
+
+PAYLOAD_MALFORMED = "The message payload is malformed"
+THREAD_ID_MISSING = "The specified Thread does not exist in the database"
+
+
+@messages_bp.route("/", methods=["POST"])
+def post_message() -> Response:
+    payload = request.get_json()
+    v = Validator(Exists("thread_id", "body", "is_from_internal", "sent_by"))
+    if not v.validate(payload):
+        logger.error(v.errors, url=request.url)
+        raise BadRequest(str(v.errors))
+
+    try:
+        response = create_message(payload)
+    except IntegrityError as e:
+        logger.error(e.params, thread_id=payload["thread_id"])
+        raise NotFound(THREAD_ID_MISSING)
```

Comment:
Probably linked to the flattened API (rather than nested resources) but it feels odd returning a 404 on the messages resource when the thing missing is the thread. While the `The specified Thread does not exist in the database` error is in the response, the 404 would usually refer to the location URL.

As the ID is not in the URL (it's in the POSTed body request) it should probably be a 400 Bad Request.